crates/duckdb-loadable-macros/src/lib.rs

```diff
@@ -1,11 +1,84 @@
 #![allow(clippy::redundant_clone)]
-use proc_macro2::{Ident, Span};
+use proc_macro2::{Ident, Literal, Punct, Span};
 
 use syn::{parse_macro_input, spanned::Spanned, Item};
 
 use proc_macro::TokenStream;
 use quote::quote_spanned;
 
+use darling::{ast::NestedMeta, Error, FromMeta};
+use syn::ItemFn;
+
+/// For parsing the arguments to the duckdb_entrypoint_c_api macro
+#[derive(Debug, FromMeta)]
+struct CEntryPointMacroArgs {
+    #[darling(default)]
+    /// The name to be given to this extension. This name is used in the entrypoint function called by duckdb
+    ext_name: String,
+    /// The minimum C API version this extension requires. It is recommended to set this to the lowest possible version
+    /// at which your extension still compiles
+    min_duckdb_version: Option<String>,
+}
+
+/// Wraps an entrypoint function to expose an unsafe extern "C" function of the same name.
+/// Warning: experimental!
+#[proc_macro_attribute]
+pub fn duckdb_entrypoint_c_api(attr: TokenStream, item: TokenStream) -> TokenStream {
+    let attr_args = match NestedMeta::parse_meta_list(attr.into()) {
+        Ok(v) => v,
+        Err(e) => {
+            return TokenStream::from(Error::from(e).write_errors());
+        }
+    };
+
+    let args = match CEntryPointMacroArgs::from_list(&attr_args) {
+        Ok(v) => v,
+        Err(e) => {
+            return TokenStream::from(e.write_errors());
+        }
+    };
+
+    // Set the minimum duckdb version (dev by default)
+    let minimum_duckdb_version = match args.min_duckdb_version {
+        Some(i) => i,
+        None => "dev".to_string(),
+    };
+
+    let ast = parse_macro_input!(item as syn::Item);
+
+    match ast {
+        Item::Fn(mut func) => {
+            let c_entrypoint = Ident::new(format!("{}_init_c_api", args.ext_name).as_str(), Span::call_site());
+            let original_funcname = func.sig.ident.to_string();
+            let prefixed_original_function = func.sig.ident.clone();
+
+            quote_spanned! {func.span()=>
+
+                /// # Safety
+                ///
+                /// Will be called by duckdb
+                #[no_mangle]
+                pub unsafe extern "C" fn #c_entrypoint(info: ffi::duckdb_extension_info, access: ffi::duckdb_extension_access) {
```

Comment:
access needs to be a `*const ffi::duckdb_extension_access` instead of a `ffi::duckdb_extension_access`

this somehow works on macos (i think due to the way arguments are passed in ARM on the stack sometimes) but breaks on x86_64 platforms in my testing. painful to debug this one

Reply:
for reference found it here https://github.com/0xcaff/duckdb_protobuf/pull/17/commits/7467ea6a31050385aabdb1108526dcaad5b9cf99

Reply:
i think this is why the windows check is failing currently